pkgs/applications/misc/input-leap/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, qttools, libei ? null, libportal,
+  mkDerivation, pkg-config, openssl, wrapGAppsHook,
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
+}:
```

Comment:
```suggestion
{ lib
, fetchFromGitHub
, cmake
, curl
, avahi
, qtbase
, qttools
, libei
, libportal
, mkDerivation
, pkg-config
, openssl
, wrapGAppsHook
, avahi
}:
```

Reply:
This suggestion seems to change the logic of `avahi` significantly, and also duplicate the `avahi` argument?

Reply:
Also `xorg` is gone so we'll have trouble importing the required libX* dependencies — I'm a bit confused as to the intention here.

Reply:
1. Sorry, the avahi duplication was not intentional.
2. `xorg.????` is deprecated. Use the X tools directly.

Reply:
> Customize avahi as you are doing should be done in top-level `all-packages.nix`.

I think the point of this idiom (it comes from the `barrier` derivation, so I can't say for sure) is to indicate that the derivation _depends_ on having libdnssd compat and won't work without it — so this is a way to indicate that in the derivation. That seems like a worthy goal to be, and I'm not sure how to indicate it otherwise.

How should I refer to e.g. `xorg.libXext` now? It doesn't exist at the top level, and this package won't build without it. Is it somewhere else in the package hierarchy? I'm not sure I understand what you mean by ‘use the X tools directly’ — I think that's what I'm doing here by adding them directly as dependencies.

Reply:
> That seems like a worthy goal to me, and I'm not sure how to indicate it otherwise.

It should be done outside the `default.nix`. More precisely, in the `callPackage` you wrote in all-packages.nix.

The expression inside default.nix shall be generalistic, and indeed "design patterns" like `package ? defaultValue` are discouraged.

>  How should I refer to e.g. xorg.libXext now?

It is automagically exported to the top level. You should declare them explicitly, as inputs.

Reply:
I guess this is something to do with `pkgs/top-level/splice.nix` — but I couldn't find any documentation or commentary indicating that `xorg` is deprecated.  Maybe we should document this in the manual/source, or throw a warning on use of `xorg`?

Reply:
@AndersonTorres — is there any documentation around this?  Should we create it?